hypervisor/arch/x86/guest/vm.c

```diff
@@ -130,6 +130,40 @@ static void create_prelaunched_vm_e820(struct acrn_vm *vm)
 	vm->e820_entry_num = E820_MAX_ENTRIES;
 	vm->e820_entries = (struct e820_entry *)ve820_entry;
 }
+
+/**
+ * @pre vm != NULL && vm_config != NULL
+ */
+static void create_ept_from_e820(struct acrn_vm *vm, const struct acrn_vm_config *vm_config)
+{
+	uint64_t base_hpa = vm_config->memory.start_hpa;
```

Comment:
Can we reuse code of normal VM? Normal VM has similar operations to do, right?

Reply:
Or the code SOS used. It also create EPT from E820.

Reply:
SOS maps all physical memory with UC attr and then go thru e820 and modify it to WB for E820_TYPE_RAM entries.  The duplicated code is not too much and it seems not easy to reuse them.

For UOS, EPT is created thru hypercall.

Reply:
Normal world would add EPT mapping through hypercall one e820 by one. While SOS will create the EPT mapping for the whole address space to UC first and then modify the DM to WB according to e820. If we plan to only create EPT mapping for DM for SOS, we could reuse this code. The different is the hpa.

Reply:
> Normal world would add EPT mapping through hypercall one e820 by one. While SOS will create the EPT mapping for the whole address space to UC first and then modify the DM to WB according to e820. If we plan to only create EPT mapping for DM for SOS, we could reuse this code. The different is the hpa.

These few functions are small enough. If we want to reuse them seems we need to add a couple of "if () else()" which may make it more complicated.

BTW, seems it make sense to change the function name to make it consistent to SOS counterpart.
create_ept_from_e820() -> prepare_prelaunched_vm_memmap()